### DISCUSSION TITLE
When does Airbyte create tables in the destination, and how does it decide the schema?

### DISCUSSION THREAD

**Discussion by rahilsondhi** [airbytehq/airbyte#28309] [Connector Questions]
Airbyte created this table in Snowflake using the wrong schema. `created_at` and `updated_at` should be timestamps, but they're varchar. Also, why did it cluster it by `_airbyte_emitted_at`? It should not be manually clustered.

<img width="919" alt="Screenshot_2023-07-13_at_5_32_55_PM" src="https://github.com/airbytehq/airbyte/assets/804316/281ee2c9-cf88-4fee-9429-0df13b58e6b8">

When I click on Source > Fields, it shows the correct schema:

<img width="959" alt="Mouse_Highlight_Overlay" src="https://github.com/airbytehq/airbyte/assets/804316/a5d94485-fc23-49be-af11-4e87654f5e72">

So I'm just confused why Airbyte created the table that way.

I tried dropping the table in Snowflake, and I triggered a new sync in Airbyte, hoping that it would re-create the table with the correct schema, but it did not, even though the sync "succeeded."

<img width="1386" alt="image" src="https://github.com/airbytehq/airbyte/assets/804316/d57ebd60-7aba-448a-ba33-324d64d01aaa">

I wanted to try the `Reset your data` button, but the [docs](https://docs.airbyte.com/operator-guides/reset/) say it will delete all the rows in my destination tables. Is that in my entire Snowflake database, or just the active streams within this connection? Because I only have one stream/table in this connection, the `webhooks` table. It's fine if Airbyte truncates that table, but I'm terrified that it will truncate all tables in the Snowflake database.

---

To summarize, my questions are:

1. Does Airbyte create tables in the destination?
   a. If so, when does it do it? Does it verify the table exists before starting a sync?
   b. How does it come up with the schema? Why did it pick varchar for timestamp columns?
   c. How does it pick the `cluster by`?
2. I deleted the Snowflake table and triggered an Airbyte sync, and it succeeded, but the table doesn't exist, so what happened?
3. Can we get more specific about what `Reset your data` does? I hope it only truncates the **active tables** for the **current connection**, not **all tables in the destination**

Thank you!

[a comment by an automated account is not shown]
